gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequestInfo.java

```diff
@@ -0,0 +1,358 @@
+      this.startTimeMillis = info.startTimeMillis;
+      this.status = info.status;
+    }
+
+    @Override
+    public Builder additions(List<RecordSet> additions) {
+      this.additions = Lists.newLinkedList(checkNotNull(additions));
+      return this;
+    }
+
+    @Override
+    public Builder deletions(List<RecordSet> deletions) {
+      this.deletions = Lists.newLinkedList(checkNotNull(deletions));
+      return this;
+    }
+
+    @Override
+    public Builder add(RecordSet recordSet) {
+      this.additions.add(checkNotNull(recordSet));
+      return this;
+    }
+
+    @Override
+    public Builder delete(RecordSet recordSet) {
+      this.deletions.add(checkNotNull(recordSet));
+      return this;
+    }
+
+    @Override
+    public Builder clearAdditions() {
+      this.additions.clear();
+      return this;
+    }
+
+    @Override
+    public Builder clearDeletions() {
+      this.deletions.clear();
+      return this;
+    }
+
+    @Override
+    public Builder removeAddition(RecordSet recordSet) {
+      this.additions.remove(recordSet);
+      return this;
+    }
+
+    @Override
+    public Builder removeDeletion(RecordSet recordSet) {
+      this.deletions.remove(recordSet);
+      return this;
+    }
+
+    @Override
+    public ChangeRequestInfo build() {
+      return new ChangeRequestInfo(this);
+    }
+
+    @Override
+    Builder id(String id) {
+      this.id = checkNotNull(id);
+      return this;
+    }
+
+    @Override
+    Builder startTimeMillis(long startTimeMillis) {
+      this.startTimeMillis = startTimeMillis;
+      return this;
+    }
+
+    @Override
+    Builder status(ChangeRequestInfo.Status status) {
+      this.status = checkNotNull(status);
+      return this;
+    }
+  }
+
+  ChangeRequestInfo(BuilderImpl builder) {
+    this.additions = ImmutableList.copyOf(builder.additions);
+    this.deletions = ImmutableList.copyOf(builder.deletions);
+    this.id = builder.id;
+    this.startTimeMillis = builder.startTimeMillis;
+    this.status = builder.status;
+  }
+
+  /**
+   * Returns an empty builder for the {@code ChangeRequestInfo} class.
+   */
+  public static Builder builder() {
+    return new BuilderImpl();
+  }
+
+  /**
+   * Creates a builder populated with values of this {@code ChangeRequestInfo}.
+   */
+  public Builder toBuilder() {
+    return new BuilderImpl(this);
+  }
+
+  /**
+   * Returns the list of {@link RecordSet}s to be added to the zone upon submitting this change
+   * request.
+   */
+  public List<RecordSet> additions() {
+    return additions;
+  }
+
+  /**
+   * Returns the list of {@link RecordSet}s to be deleted from the zone upon submitting this change
+   * request.
+   */
+  public List<RecordSet> deletions() {
+    return deletions;
+  }
+
+  /**
+   * Returns the id assigned to this {@code ChangeRequest} by the server.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this {@code ChangeRequest} was started by the server.
```

Comment:
Do you think it is worth mentioning that value would be `null` when change request was not applied (status == pending)?